app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt

```diff
@@ -0,0 +1,322 @@
+                  OutlinedTextField(
+                      value = (exerciseDetail as ExerciseDetail.TimeBased).sets.toString(),
+                      onValueChange = {
+                        val newValue = it.toIntOrNull()
+                        if (newValue != null) {
+                          exerciseDetail =
+                              (exerciseDetail as ExerciseDetail.TimeBased).copy(sets = newValue)
+                        }
+                      },
+                      label = { Text("Sets") },
+                      modifier = Modifier.testTag("setsTextField"))
+                }
+                is ExerciseDetail.RepetitionBased -> {
+                  OutlinedTextField(
+                      value =
+                          (exerciseDetail as ExerciseDetail.RepetitionBased).repetitions.toString(),
+                      onValueChange = {
+                        val newValue = it.toIntOrNull()
+                        if (newValue != null) {
+                          exerciseDetail =
+                              (exerciseDetail as ExerciseDetail.RepetitionBased).copy(
+                                  repetitions = newValue)
+                        }
+                      },
+                      label = { Text("Repetitions") },
+                      modifier = Modifier.testTag("repetitionsTextField"))
+                }
+                else -> {}
+              }
+            }
+          }
+        },
+        confirmButton = {
+          Button(
+              onClick = {
+                if (selectedExerciseType != null && exerciseDetail != null) {
+                  exerciseList =
+                      exerciseList +
+                          when (workoutType) {
+                            WorkoutType.YOGA ->
+                                YogaExercise(
+                                    exerciseId = workoutViewModel.getNewUid(),
+                                    type = selectedExerciseType as YogaExerciseType,
+                                    detail = exerciseDetail!!)
+                            WorkoutType.BODY_WEIGHT ->
+                                BodyWeightExercise(
+                                    exerciseId = workoutViewModel.getNewUid(),
+                                    type = selectedExerciseType as BodyWeightExerciseType,
+                                    detail = exerciseDetail!!)
+                            else -> throw IllegalArgumentException("Unsupported workout type")
+                          }
+                  showExerciseDialog = false
+                }
+              },
+              modifier = Modifier.testTag("addExerciseConfirmButton")) {
+                Text("Add")
+              }
+        },
+        dismissButton = {
+          Button(
+              onClick = { showExerciseDialog = false },
+              modifier = Modifier.testTag("addExerciseCancelButton")) {
+                Text("Cancel")
+              }
+        })
+  }
+}
```

Comment:
As you said, this screen needs work to make it more attractive.

Try annotating your code to make it easier to read.